docs/griptape-framework/structures/task-memory.md

```diff
@@ -1,89 +1,345 @@
+[04/30/24 16:36:52] INFO     Subtask 61ec822bfe49472c9ed874fad07d13a1
+                             Thought: First, I need to scrape the content of the provided URL. Then, I will search the scraped content for the number of copies of Elden Ring that have been sold. Finally, I will save this information to a file.
+
+                             Actions: [{"name": "WebScraper", "path": "get_content", "input": {"values": {"url": "https://en.wikipedia.org/wiki/Elden_Ring"}}, "tag": "scrape_elden_ring"}]
+[04/30/24 16:37:04] INFO     Subtask 61ec822bfe49472c9ed874fad07d13a1
+                             Response: Output of "WebScraper.get_content" was stored in memory with memory_name "TaskMemory" and artifact_namespace "c7a01e8202e24869b7be559e0daff110"
+[04/30/24 16:37:10] INFO     Subtask 32f9cb73d8944e5ca36bd68a90e4f4b2
+                             Thought: Now that the webpage content is stored in memory, I need to query this memory to find the number of copies of Elden Ring that have been sold.
+                             Actions: [{"tag": "query_sales", "name": "TaskMemoryClient", "path": "query", "input": {"values": {"memory_name": "TaskMemory", "artifact_namespace": "c7a01e8202e24869b7be559e0daff110", "query": "How many copies of Elden Ring have been sold?"}}}]
+[04/30/24 16:37:18] INFO     Subtask 32f9cb73d8944e5ca36bd68a90e4f4b2
+                             Response: Output of "TaskMemoryClient.query" was stored in memory with memory_name "TaskMemory" and artifact_namespace "f8dd40fb302a47d7862c8b76eeaf61c2"
+[04/30/24 16:37:25] INFO     Subtask 74a5fd392b044956842a56b76d09183e
+                             Thought: Now that I have the number of copies sold stored in memory, I need to save this information to a file.
+                             Actions: [{"tag": "save_sales", "name": "FileManager", "path": "save_memory_artifacts_to_disk", "input": {"values": {"dir_name": "sales_data", "file_name": "elden_ring_sales.txt", "memory_name": "TaskMemory", "artifact_namespace": "f8dd40fb302a47d7862c8b76eeaf61c2"}}}]
+                    INFO     Subtask 74a5fd392b044956842a56b76d09183e
+                             Response: Successfully saved memory artifacts to disk
+[04/30/24 16:37:27] INFO     ToolkitTask 3d3c5f5a98a44f32ad9533621c036604
+                             Output: The number of copies of Elden Ring sold has been successfully saved to the file "elden_ring_sales.txt" in the "sales_data" directory.
 ```
+
+## Tools That Can Read From Task Memory
+
+As seen in the previous example, certain Tools are designed to read directly from Task Memory. This means that you can use these Tools to interact with the data stored in Task Memory without needing to pass it through the LLM.
+
+Unless you're handing off data to another Tool capable of reading from Task Memory, you should most likely set `off_prompt` to `False` on these Tools otherwise you may end up [Task Memory Looping](#task-memory-looping).
```

Comment:
nit: reword slightly to lean into the default being False now.

> Unless you're handing off data to another Tool capable of reading from Task Memory, you should most likely keep `off_prompt` set to `False` on these Tools

Reply:
Removed, we already have a section that calls out Task Memory Looping. And this behavior is now harder to accidentally stumble into.